cromwell-drs-localizer/src/main/scala/drs/localizer/downloaders/GetmChecksum.scala

```diff
@@ -1,35 +1,57 @@
 package drs.localizer.downloaders
 
+import cats.syntax.validated._
 import cloud.nio.impl.drs.AccessUrl
+import common.validation.ErrorOr.ErrorOr
 import drs.localizer.downloaders.AccessUrlDownloader.Hashes
+import mouse.all.anySyntaxMouse
+import org.apache.commons.codec.binary.Base64.{decodeBase64, isBase64}
+import org.apache.commons.codec.binary.Hex.encodeHexString
 import org.apache.commons.text.StringEscapeUtils
 
+
 sealed trait GetmChecksum {
   def getmAlgorithm: String
-  def value: String
-  def args: String = {
+  def rawValue: String
+  def value: ErrorOr[String] = rawValue.validNel
+  def args: ErrorOr[String] = {
     // The value for `--checksum-algorithm` is constrained by the algorithm names in the `sealed` hierarchy of
     // `GetmChecksum`, but the value for `--checksum` is largely a function of data returned by the DRS server.
     // Shell escape this to avoid injection.
-    val escapedValue = StringEscapeUtils.escapeXSI(value)
-    s"--checksum-algorithm '$getmAlgorithm' --checksum $escapedValue"
+    value map { v =>
+      val escapedValue = StringEscapeUtils.escapeXSI(v)
+      s"--checksum-algorithm '$getmAlgorithm' --checksum $escapedValue"
+    }
   }
 }
-case class Md5(override val value: String) extends GetmChecksum {
+
+case class Md5(override val rawValue: String) extends GetmChecksum {
+  override def value: ErrorOr[String] = {
+    val trimmed = rawValue.trim
+    if (trimmed.matches("[A-Fa-f0-9]+"))
+      trimmed.validNel
+    // Azure reports its md5s in base64, but `getm` knows only hex. For the sanity of humans storing data in Azure
+    // it's probably best if the DRS localizer does this conversion for `getm`.
+    else if (isBase64(trimmed))
+      (trimmed |> decodeBase64 |> encodeHexString).validNel
```

Comment:
I'd rather we not have this in the code, but that depends on https://broadworkbench.atlassian.net/browse/DR-2259 getting done. I'm thinking of adding that link to the comment above and creating a ticket on our board to remove this condition. Thoughts?

Reply:
Sounds like a good plan! I agree that we should remove this eventually but don't want it to block this merge.